examples/black_scholes_greeks.py

```diff
@@ -0,0 +1,217 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import argparse
+from enum import IntEnum
+
+from benchmark import CuNumericTimer, parse_args, run_benchmark
+
+NUM_ITERS = 10
+WARMUP_ITER = 2
+
+vol_start = 0.1
+vol_step = 0.01
+t_start = 0.5
+t_step = 1.0 / (365 * 10)
+money_start = -0.4
+money_step = 0.01
+
+
+RISKFREE = 0.02
+S0 = 100.0
+N_GREEKS = 7
+
+
+class Greeks(IntEnum):
+    PREM = (0,)
+    DELTA = (1,)
+    VEGA = (2,)
+    GAMMA = (3,)
+    VANNA = (4,)
+    VOLGA = (5,)
+    THETA = 6
+
+
+def initialize(n_vol_steps, n_t_steps, n_money_steps, D):
+    CALL = np.zeros(
+        (
+            N_GREEKS,
+            n_t_steps,
+            n_vol_steps,
+            n_money_steps,
+        ),
+        dtype=D,
+    )
+    PUT = np.zeros(
+        (
+            N_GREEKS,
+            n_t_steps,
+            n_vol_steps,
+            n_money_steps,
+        ),
+        dtype=D,
+    )
+    S = np.full(
+        (
+            n_t_steps,
+            n_vol_steps,
+            n_money_steps,
+        ),
+        S0,
+        dtype=D,
+    )
+    temp_arr = np.arange((n_vol_steps * n_t_steps * n_money_steps), dtype=int)
+    k_temp = (temp_arr % n_money_steps) * money_step
+    k_temp = k_temp.reshape(
+        (
+            n_t_steps,
+            n_vol_steps,
+            n_money_steps,
+        )
+    )
+    K = (k_temp + (1 + money_start)) * S0
+
+    t_temp = (temp_arr % (n_vol_steps * n_money_steps)) * vol_step
+    t_temp = t_temp.reshape(
+        (
+            n_t_steps,
+            n_vol_steps,
+            n_money_steps,
+        )
+    )
+    T = t_temp + t_start
+    R = 0.02
+    V = np.full((n_t_steps, n_vol_steps, n_money_steps), vol_start, dtype=D)
+    for i in range(n_vol_steps):
+        V[:, i, :] += i * vol_step
+
+    return CALL, PUT, S, K, T, R, V
+
+
+# Cumulative distribution function
+# https://en.wikipedia.org/wiki/Cumulative_distribution_function
+def normCDF(d):
+    A1 = 0.31938153
+    A2 = -0.356563782
+    A3 = 1.781477937
+    A4 = -1.821255978
+    A5 = 1.330274429
+    RSQRT2PI = 0.39894228040143267793994605993438
+
+    K = 1.0 / (1.0 + 0.2316419 * np.absolute(d))
+
+    cnd = (
+        RSQRT2PI
+        * np.exp(-0.5 * d * d)
+        * (K * (A1 + K * (A2 + K * (A3 + K * (A4 + K * A5)))))
+    )
+
+    return np.where(d > 0, 1.0 - cnd, cnd)
+
+
+# Probability density function.
+# https://en.wikipedia.org/wiki/Probability_density_function
+def normPDF(d):
+    RSQRT2PI = 0.39894228040143267793994605993438
```

Comment:
`RSQRT2PI` is used in both `normCDF` and `normPDF` normally I would suggest moving this out to the top level. Since these are not called in a tight loop (for a given call to `black_schole`) I don't think it would noticeably affect any benchmark (though that could be checked)